interactive/submit.py

```diff
@@ -50,18 +53,70 @@ def submit_analysis(
         force=force,
     )
 
-    job_request = project.interactive_workspace.job_requests.create(
+    create_job_request(analysis_request, backend, sha, project_yaml)
+
+    analysis_request.save()
+    notify_analysis_request_submitted(analysis_request)
+
+    return analysis_request
+
+
+def create_job_request(analysis_request, backend, sha, project_yaml):
+    workspace = analysis_request.project.interactive_workspace
+    job_request = workspace.job_requests.create(
         backend=backend,
-        created_by=creator,
+        created_by=analysis_request.created_by,
         sha=sha,
         project_definition=project_yaml,
         force_run_dependencies=True,
         requested_actions=["run_all"],
     )
-
     analysis_request.job_request = job_request
     analysis_request.save()
 
-    notify_analysis_request_submitted(analysis_request)
 
-    return analysis_request
+def resubmit_analysis(analysis_request, github_api=None):
+    if github_api is None:  # noqa
+        github_api = _get_github_api()
+    repo_url = analysis_request.project.interactive_workspace.repo.url
+    sha, project_yaml = get_existing_commit(analysis_request.id, repo_url, github_api)
+    create_job_request(
+        analysis_request, analysis_request.job_request.backend, sha, project_yaml
+    )
+
+
+def get_existing_commit(analysis_id, repo_url, github_api):
+    if repo_url.startswith("http"):
+        return _get_existing_commit_github(analysis_id, repo_url, github_api)
+    else:  # support local repositories for testing
+        return _get_existing_commit_local(analysis_id, repo_url)
+
+
+def _get_existing_commit_github(analysis_id, repo_url, github_api):
+    """Used for github repos, in production."""
+    url = furl(repo_url)
+    org, repo = url.path.segments[:2]
```

Comment:
If you pass a Repo instance to this function instead of the URL, it's got the `name` and `owner` properties to do this for you :)